cloud/scope/powervs_cluster_test.go

```diff
@@ -153,3 +159,1432 @@ func TestNewPowerVSClusterScope(t *testing.T) {
+		g.Expect(err).To(BeNil())
+		clusterScope.IBMPowerVSCluster.Spec.ResourceGroup = &infrav1beta2.IBMPowerVSResourceReference{ID: ptr.To("resourceGroupID")}
+
+		requeue, err := clusterScope.ReconcilePowerVSServiceInstance()
+		g.Expect(err).To(BeNil())
+		g.Expect(requeue).To(BeTrue())
+		g.Expect(*clusterScope.IBMPowerVSCluster.Status.ServiceInstance.ID).To(Equal("instance-GUID"))
+		g.Expect(*clusterScope.IBMPowerVSCluster.Status.ServiceInstance.ControllerCreated).To(BeTrue())
+	})
+}
+
+func TestCheckServiceInstanceState(t *testing.T) {
+	testCases := []struct {
+		name        string
+		requeue     bool
+		expectedErr error
+		instance    resourcecontrollerv2.ResourceInstance
+	}{
+		{
+			name:     "Service instance is in active state",
+			instance: resourcecontrollerv2.ResourceInstance{Name: ptr.To("instance"), State: ptr.To("active")},
+		},
+		{
+			name:     "Service instance is in provisioning state",
+			instance: resourcecontrollerv2.ResourceInstance{Name: ptr.To("instance"), State: ptr.To("provisioning")},
+			requeue:  true,
+		},
+		{
+			name:        "Service instance is in failed state",
+			instance:    resourcecontrollerv2.ResourceInstance{Name: ptr.To("instance"), State: ptr.To("failed")},
+			expectedErr: fmt.Errorf("PowerVS service instance is in failed state"),
+		},
+		{
+			name:        "Service instance is in unknown state",
+			instance:    resourcecontrollerv2.ResourceInstance{Name: ptr.To("instance"), State: ptr.To("unknown")},
+			expectedErr: fmt.Errorf("PowerVS service instance is in unknown state"),
+		},
+	}
+
+	for _, tc := range testCases {
+		g := NewWithT(t)
+		t.Run(tc.name, func(_ *testing.T) {
+			clusterScope := PowerVSClusterScope{}
+			requeue, err := clusterScope.checkServiceInstanceState(tc.instance)
+			g.Expect(requeue).To(Equal(tc.requeue))
+			if tc.expectedErr != nil {
+				g.Expect(err).To(Equal(tc.expectedErr))
+			} else {
+				g.Expect(err).To(BeNil())
+			}
+		})
+	}
+}
+
+func TestIsServiceInstanceExists(t *testing.T) {
+	var (
+		mockResourceController *mockRC.MockResourceController
+		mockCtrl               *gomock.Controller
+	)
+	setup := func(t *testing.T) {
+		t.Helper()
+		mockCtrl = gomock.NewController(t)
+		mockResourceController = mockRC.NewMockResourceController(mockCtrl)
+	}
+	teardown := func() {
+		mockCtrl.Finish()
+	}
+	t.Run("When get service instance returns error", func(t *testing.T) {
+		g := NewWithT(t)
+		setup(t)
+		t.Cleanup(teardown)
+
+		clusterScopeParams := getPowerVSClusterScopeParams()
+		clusterScopeParams.ResourceControllerFactory = func() (resourcecontroller.ResourceController, error) {
+			mockResourceController.EXPECT().GetServiceInstance(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, errors.New("failed to get service instance"))
+			return mockResourceController, nil
+		}
+		clusterScope, err := NewPowerVSClusterScope(clusterScopeParams)
+		g.Expect(err).To(BeNil())
+
+		instance, requeue, err := clusterScope.isServiceInstanceExists()
```

Comment:
```suggestion
		instanceID, requeue, err := clusterScope.isServiceInstanceExists()
```
please change this wherever applicable in this test case

Reply:
Updated at all the places.